### PR TITLE
[DMP-566] Add button to open Ventana Banorte

### PR DESCRIPTION
**Pull request by markriverac**
## What is done

[[DMP-566}] Importa scripts para abrir la ventana de pagos de Banorte](https://deacero.atlassian.net/browse/DMP-566)

## How is done

1. Created a new hook to use scripts in components.
2. Created a new component that implements the Button component.
3. Use the new hook to load the scripts for the integration with Banorte.
4. Starts the process to open the payment window.

**Note: This PR still has many errors, these are related to Banorte Documentation. They will be solved in the following tickets (I hope so)**

## How to test

1. Run the project via `npm start`
2. Go to the `carrito/pago` page.
3. Click on the "Realiza tu pedido y paga" button.
4. You should view the Banorte Popup displaying an error. (Probably you will need to click the button multiple times).

### CODE REVIEW

File: src/components/common/PayButton/PayButton.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    //     // TO DO Code Here when an error occurred
+    //   },
+    //   onSuccess: function () {
+    //     // TO DO Code Here when operation is successful
+    //   },
+    //   onCancel: function () {
+    //     // TO DO Code Here when user cancel operation
+    //   },
+    // });
+  };
```
Comment (dianabasilio) [inline src/components/common/PayButton/PayButton.tsx:59]: Will these comments be used in the future and for what? or not?

Comment (markriverac) [reply src/components/common/PayButton/PayButton.tsx:59]: Thank you, these aren't necessary. I will remove them

File: src/components/hooks/useScript.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+
+export type Status = "idle" | "loading" | "ready" | "error";
+export type ScriptElt = HTMLScriptElement | null;
+
+function useScript(src: string): Status {
+  const [status, setStatus] = useState<Status>(src ? "loading" : "idle");
+
```
[the remark this answers is not in the document]
Comment (markriverac) [reply src/components/hooks/useScript.tsx:8]: `idle` is used for something like being on standby, lazy, or doing nothing. In this scenario, when you pass an empty string, you will load the hook, but at the beginning will do nothing.

**Review by dianabasilio — APPROVED** [review]

File: src/components/common/PayButton/PayButton.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const PayButton = () => {
+  useScript(
+    `https://multicobros.banorte.com/orquestador/resources/js/jquery-3.3.1.js`
+  );
+  useScript(`https://multicobros.banorte.com/orquestador/lightbox/checkout.js`);
+
+  const openBanorteWindow = () => {
+    if (typeof Payment !== "undefined") {
+      console.log(Payment);
+      Payment.setEnv("pro");
```
Comment (dianabasilio) [inline src/components/common/PayButton/PayButton.tsx:15]: could you delete console log?

**Pull request merged**